### PR TITLE
add tests for system_hooks

### PR DESCRIPTION
**Pull request by venky333**
* adds tests for system_hooks

### CODE REVIEW

File: system_hooks_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+	mux.HandleFunc("/api/v4/hooks", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		fmt.Fprint(w, `[{"id":1,"url":"https://gitlab.example.com/hook"}]`)
+	})
+	hooks, _, err := client.SystemHooks.ListHooks()
+
+	require.NoError(t, err)
+	want := []*Hook{{ID: 1, URL: "https://gitlab.example.com/hook"}}
+	require.Equal(t, want, hooks)
```
Comment (svanharmelen) [inline system_hooks_test.go:23]:
```suggestion

	hooks, _, err := client.SystemHooks.ListHooks()
	require.NoError(t, err)
	
	want := []*Hook{{ID: 1, URL: "https://gitlab.example.com/hook"}}
	require.Equal(t, want, hooks)
```

File: system_hooks_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	})
+
+	opt := &AddHookOptions{
+		URL: String("https://gitlab.example.com/hook"),
+	}
+	h, _, err := client.SystemHooks.AddHook(opt)
+
+	require.NoError(t, err)
+	want := &Hook{ID: 1, URL: "https://gitlab.example.com/hook", CreatedAt: (*time.Time)(nil)}
+	require.Equal(t, want, h)
```
Comment (svanharmelen) [inline system_hooks_test.go:42]:
```suggestion

	h, _, err := client.SystemHooks.AddHook(opt)
	require.NoError(t, err)
	
	want := &Hook{ID: 1, URL: "https://gitlab.example.com/hook", CreatedAt: (*time.Time)(nil)}
	require.Equal(t, want, h)
```

File: system_hooks_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+	mux.HandleFunc("/api/v4/hooks/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodGet)
+		fmt.Fprint(w, `{"project_id" : 1,"owner_email" : "example@gitlabhq.com","owner_name" : "Someone",
+				"name" : "Ruby","path" : "ruby","event_name" : "project_create"}`)
+	})
+
+	he, _, err := client.SystemHooks.TestHook(1)
+
+	require.NoError(t, err)
```
Comment (svanharmelen) [inline system_hooks_test.go:57]:
```suggestion
	require.NoError(t, err)
	
```

File: system_hooks_test.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+	mux.HandleFunc("/api/v4/hooks/1", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, http.MethodDelete)
+	})
+
+	_, err := client.SystemHooks.DeleteHook(1)
+	require.NoError(t, err)
+}
+
+func TestHook_String(t *testing.T) {
```
Comment (svanharmelen) [inline system_hooks_test.go:81]: Let's not test this as this functionally has no value and you already added test for the `Stringify` function.

File: system_hooks_test.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+		ID:        1,
+		URL:       "https://github.com/",
+		CreatedAt: nil,
+	}
+	hs := h.String()
+	want := "gitlab.Hook{ID:1, URL:\"https://github.com/\"}"
+	require.Equal(t,want , hs)
+}
+
+func TestHookEvent_String(t *testing.T) {
```
Comment (svanharmelen) [inline system_hooks_test.go:92]: Let's not test this as this functionally has no value and you already added test for the `Stringify` function.

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: system_hooks_test.go (hunk 4 above)
Comment (venky333) [reply system_hooks_test.go:83]: done

File: system_hooks_test.go (hunk 5 above)
Comment (venky333) [reply system_hooks_test.go:94]: done

**Review by svanharmelen — APPROVED** [review]
Thanks @venky333 👍🏻

**Pull request merged**